### ISSUE TITLE
.npmignore conflicts with package.json

### ISSUE THREAD

**Issue by hallelujah** [videojs/video.js#3172]
## Description
When using github as the source in my npm package.json, it will lead with no usable js files
Only the src/css files are present.

## Steps to reproduce

* Create a new package.json with this content

```
{
  "name": "video",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "dependencies": {
  },
  "devDependencies": {
    "video.js": "git@github.com:videojs/video.js.git#master"
  },
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC"
}
```

* Install with `npm install`

### Expected
We should at least have either src/js/** or dist/**

### Actual
Only src/css is fetched

### versions
Current master branch 0b26d83e9981560f641f70feb270d6ddcc49ec41

I think the problem is because of the use of .npmignore and package.json 

In .npmignore it excludes everything but dist and src/css

```
# Exclude everything but the contents of the dist directory.
**/*
!dist/**
!src/css/**
```

In package.json it ignores dist too

```json
 "standard": {
    "ignore": [
      "**/Gruntfile.js",
      "**/build/**",
      "**/dist/**",
      "**/docs/**",
      "**/lang/**",
      "**/sandbox/**",
      "**/test/**"
    ]
  }
```

**Comment by gkatsev** [comment]
The files you would want to be using with videojs (right now) are the ones in `dist/`.
We're tracking making the main sources available in https://github.com/videojs/video.js/issues/2750.
The `standard: ignore` thing is for our linting system.
Would you happen to be using browserify or webpack?

**Comment by hallelujah** [comment]
Hello @gkatsev thanks for your quick answer,

No I don't use neither browserify or webpack (for now), I just begin a small project while keeping all dependencies in package.json and referencing to the most up to date source of videojs (say github, I do not want to downlad tarball nor have git submodules, npm should be enough I think)

I think that it should be convenient to reference videojs as a git url in package.json
Sadly with the rules above applied when `npm install` there is no js at all.

**Comment by gkatsev** [comment]
Ah, I see. Why are you pointing videojs at the git repo? Also, if you point at a specific version, it should have the `dist` file there.

**Comment by hallelujah** [comment]
Well, it is a very new project so I would like to keep dependencies up to date with the master branch as much as possible. Pointing to a git repo should be feasible right?

Thanks, as you mentioned git@github.com:videojs/video.js.git#v5.8.3" works

But how can I point to master then (yes I like danger) ? Or is it your policy to not allow pointing to master directly since it would not contain dist, in this case I understand and I will follow the rules but it is quite frustrating :)

If so why not making src/** available but I think this is more related to #2750? I will check this discussion.

**Comment by gkatsev** [comment]
Yeah, we specifically don't check in dist into the repo. If you want the latest and greatest, you'll want to clone the repo and link it in.
Closing since we aren't going to be making any changes in videojs for this but we can continue the discussion here.
